packages/previous/index.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useRef } from 'react'
+
+/**
+ * @param value The value to be returned on the next render.
+ * @returns Returns the last value, undefined for initial render.
+ */
+export default function usePrevious<T>(value: T): T {
+  const ref = useRef<T>()
+
+  useEffect(() => {
+    ref.current = value
+  })
```

Comment:
Should we add `value` in the dep array?

Reply:
The example in the React doc does not include a deps array. I think this is fine, as we want whatever value is passed into the hook, this could include a value that has been mutated in some way so may not appear as changed. What do you think?

Reply:
Yep, that makes sense. It should be re-run on every render anyway 💯